crmprtd/download_cache_process.py

```diff
@@ -0,0 +1,371 @@
+        "dfo_ccg_lighthouse",
+    ],
+    "ytnt": [
+        "nt_forestry",
+        "nt_water",
+        "yt_gov",
+        "yt_water",
+        "yt_avalanche",
+        "yt_firewx",
+    ],
+}
+
+network_alias_names = tuple(network_aliases.keys())
+
+
+def check_network_name(network_name):
+    if network_name not in NETWORKS:
+        raise ValueError(f"Network name '{network_name}' is not recognized.")
+
+
+def log_filename(
+    network_name: str = None,
+    tag: str = None,
+    frequency: str = None,
+    province: str = None,
+    **_,
+):
+    """Return log filename (filepath). It depends on several parameters, starting
+    with the network. Its pattern is similar, but not identical, to that of the cache
+    filename.
+
+    :param network_name: Network name to include in filename.
+    :param tag: Tag to include in filename.
+    :param frequency: Download frequency to include in filename (network ec only).
+    :param province: Province code (2 letters) to include in filename (network ec only).
+    :param _: Ignored additional kw args.
+    :return: Filename.
+    """
+    check_network_name(network_name)
+    if network_name in ("ec",):
+        return f"~/{network_name}/logs/{tag}_{province.lower()}_{frequency}_json.log"
+    if network_name in ("wmb",):
+        # Oh, yes, this one has to be a special snowflake.
+        return f"~/{network_name}/logs/{tag}_mof_json.log"
+    return f"~/{network_name}/logs/{tag}_{network_name}_json.log"
+
+
+def cache_filename(
+    timestamp: datetime.datetime = datetime.datetime.now(),
+    timestamp_format: str = "%Y-%m-%dT%H:%M:%S",
+    network_name: str = None,
+    tag: str = None,
+    frequency: str = None,
+    province: str = None,
+    **_,
+) -> str:
+    """Return cache filename (filepath). It depends on several parameters, starting
+    with the network. Its pattern is similar, but not identical, to that of the log
+    filename.
+
+    :param timestamp: Timestamp to include in filename.
+    :param timestamp_format: Format for converting timestamp to string.
+    :param network_name: Network name to include in filename.
+    :param tag: Tag to include in filename.
+    :param frequency: Download frequency to include in filename (network ec only).
+    :param province: Province code (2 letters) to include in filename (network ec only).
+    :param _: Ignored additional kw args.
+    :return: Filename.
+    """
+    check_network_name(network_name)
+    ts = timestamp.strftime(timestamp_format)
+    if network_name in ("ec",):
+        return (
+            f"~/{network_name}/download/{tag}_{frequency}_{province.lower()}_{ts}.xml"
+        )
+    if network_name in (
+        "bc_env_snow",
+        "bc_forestry",
+        "bc_tran",
+        "dfo_ccg_lighthouse",
+        "nt_forestry",
+        "nt_water",
+        "yt_gov",
+        "yt_water",
+        "yt_avalanche",
+        "yt_firewx",
+    ):
+        return f"~/{network_name}/cache/{tag}_{network_name}_{ts}.xml"
```

Comment:
@jameshiebert , if we were to settle on either `cache/` or `download/` as a directory path component, that would simplify this function considerably. Would there be any adverse impact to doing that? Once we deployed this version, we'd have to rename the directories on server crmprtd. And update `infill_all`.